voluncheer/opportunityboard/views/search.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from django.shortcuts import render
+
+from opportunityboard.models import Category
+from opportunityboard.models import Opportunity
+from opportunityboard.models import Subcategory
+from opportunityboard.models import Subsubcategory
+from opportunityboard.views.opportunityboard import category_dict_gen
+
+
+def filter_search(request):
+    if request.method == "POST":
+        filter = parse_search_filter(request.POST)
+        logger = logging.getLogger(__name__)
+        logger.error(filter)
+
+    opportunity_lists = Opportunity.objects.order_by("-pubdate"[:20])
+    cate_output_dict = category_dict_gen()
+    durations = {
+        "One-day": ["<=2 Hours", "2-4 Hours", "Full Day"],
+    }
+    context = {
+        "opportunity_lists": opportunity_lists,
+        "categories": cate_output_dict,
+        "durations": durations,
+    }
+    return render(request, "voluncheer/opportunityboard.html", context)
+
+
+def parse_search_filter(post):
+    """Check if the input filters are valid:
+    category(default empty string)
+    duration(default empty string)
+    distance(default 0)
+    """
+    output = {}
+    try:
+        _, category = category_is_valid(post["category"])
+        output["category"] = category
+    except KeyError:
+        output["category"] = ""
+        pass
+    return output
+
+
+def category_is_valid(value):
+    """Check if the category is valid
+    Input: value[category_name] -> string
+    Return: error(None), target_object
+    """
+    category = Category.objects.filter(name=value).first()
+    if not category:
+        return None, category
```

Comment:
This means that we are always returning a falsey `target_object`, right? Is that desirable? Maybe I don't understand what this function is trying to do.

Reply:
Just an error handling when we can't find a match in Category, Subcategory, and Subsubcategory.